compiler/src/dotty/tools/dotc/semanticdb/ExtractSemanticDB.scala

```diff
@@ -48,12 +61,103 @@ class ExtractSemanticDB extends Phase:
 
   override def run(using Context): Unit =
     val unit = ctx.compilationUnit
-    val extractor = Extractor()
-    extractor.extract(unit.tpdTree)
-    ExtractSemanticDB.write(unit.source, extractor.occurrences.toList, extractor.symbolInfos.toList, extractor.synthetics.toList)
+    if (phaseMode == ExtractSemanticDB.PhaseMode.PostTyper)
+      val extractor = ExtractSemanticDB.Extractor()
+      extractor.extract(unit.tpdTree)
+      ExtractSemanticDB.write(unit.source, extractor.occurrences.toList, extractor.symbolInfos.toList, extractor.synthetics.toList)
+    else
+      val warnings = ctx.reporter.allWarnings.collect {
```

Comment:
this could probably be more efficient by overriding `runOn`, grouping the warnings by source file, and then running each group on the associated compilation unit

Reply:
done :+1: https://github.com/lampepfl/dotty/pull/17835/commits/d2b54c2fa3065f64c698f6625d07be818d0a8ff3